packages/x-charts/src/internals/plugins/featurePlugins/useChartKeyboardNavigation/useChartKeyboardNavigation.selectors.ts

```diff
@@ -98,9 +104,20 @@ export const selectorChartsKeyboardYAxisIndex = createSelector(
 export const selectorChartsKeyboardItem = createSelector(
   [selectorChartsFocusedSeriesType, selectorChartsFocusedSeriesId, selectorChartsFocusedDataIndex],
   function selectorChartsKeyboardItem(seriesType, seriesId, dataIndex) {
-    if (seriesId === undefined) {
+    if (seriesType === undefined || seriesId === undefined) {
       return null;
     }
-    return { seriesId, dataIndex: seriesType === 'line' ? undefined : dataIndex };
+    return {
+      type: seriesType,
+      seriesId,
+      dataIndex,
+    } as BarItemIdentifier | LineItemIdentifier | ScatterItemIdentifier | PieItemIdentifier;
```

Comment:
Could we avoid casting and either use `satisfies` or adjust the function's return type?

Reply:
I rewrote the type with dedicate union of series supported by focus. Should be cleaner now